### PR TITLE
Fix/result manager i24

### LINKED ISSUE #24: Numburn samples has no sensible default / no documentation for recommended values
**Describe the bug**
MCMC with a even number of walkers and and e.g. occurence=2 ignores half of the walkers.

Inspect load_sim_results for correct implementation (order of loaded samples etc.)
also for loading dense grid results
...

### PR DESCRIPTION
**Pull request by castellinilinguini**
# Description

Fixes #24 by modifying the way thinning and burning is performed in the result manager.

## Type of change

Please tick at least one of the options. Delete the other options.

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- [x] All unit tests pass locally.

## Checklist For Contributor

- [x] My code follows the style guidelines of this project (I installed pre-commit)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I wrote my changes in the changelog in the `[unreleased]` section
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

## Checklist For Maintainers

- [x] Continuous Integration (CI) is successfully running
- [x] Do we want to release/tag a new version? [✔️ / ❌]
  - [ ] If yes, add a release to the changelog and set the new version in pyproject.toml. After the merge, tag the release!

### CODE REVIEW

File: eulerpi/core/result_manager.py [hunk 1]
```diff
@@ -428,37 +431,43 @@ def load_slice_inference_results(
                 ndmin=2,
             )
             if i == 0:
-                overall_density_evals = density_evals
-                overall_sim_results = sim_results
-                overall_params = params
+                param_dim = params.shape[1]
+                data_dim = sim_results.shape[1]
+                overall_density_evals = density_evals.reshape(
```
Comment (kaiserls) [inline eulerpi/core/result_manager.py:436]: What will -1 resolve to? If we already know the dimension, I think we should use the dimension.
You could also get the number of steps here, if we know the data / param / density dimension.

**Review by kaiserls — CHANGES_REQUESTED** [review]
Nice work! One bug needs to be fixed and I added a few comments where I see room for improvement. But fixing #24 should be the priority.
Could you maybe write a test whether the thinning is working as intended? I think it is currently not tested. You could do it in test_inference.

File: eulerpi/core/result_manager.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaiserls) [reply eulerpi/core/result_manager.py:436]: (By using -1 for the number of steps)

File: eulerpi/core/result_manager.py [hunk 2]
```diff
@@ -411,6 +411,9 @@ def load_slice_inference_results(
         if thinning_factor is None:
             thinning_factor = inference_information["thinning_factor"]
 
+        num_steps = inference_information["num_steps"]
```
[the remark this answers is not in the document]
Comment (castellinilinguini) [reply eulerpi/core/result_manager.py:414]: Right now, I don't see how the number of steps could depend on the run - is this a feature we might want to add in the future?

File: eulerpi/core/result_manager.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (castellinilinguini) [reply eulerpi/core/result_manager.py:436]: Agreed & fixed!

File: eulerpi/core/result_manager.py [hunk 3]
```diff
@@ -428,37 +431,43 @@ def load_slice_inference_results(
...
                 )
                 overall_sim_results = np.concatenate(
-                    (overall_sim_results, sim_results)
+                    (
+                        overall_sim_results,
+                        sim_results.reshape(num_steps, num_walkers, -1),
+                    )
+                )
+                overall_params = np.concatenate(
+                    (overall_params, params).reshape(
```
[the remark this answers is not in the document]
Comment (castellinilinguini) [reply eulerpi/core/result_manager.py]: Yep, that's a typo. Should've tested it before pushing.

File: eulerpi/core/result_manager.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kaiserls) [reply eulerpi/core/result_manager.py:414]: The use case I see is the following, e.g. in a jupyter notebook:

```python
inference(...)
plt.plot(...)
```
If the results are not yet converged, start another run and plot again.

```python
inference(...)
plt.pltot(...)
```
But I really dont mind if we include this now or not.

**Review by kaiserls — APPROVED** [review]
Fixed the last problem for you :)

**Pull request merged**